openhcl/underhill_attestation/src/lib.rs

```diff
@@ -241,20 +259,26 @@ pub async fn initialize_platform_security(
     suppress_attestation: bool,
     driver: LocalDriver,
 ) -> Result<PlatformAttestationData, Error> {
+    tracing::info!(CVM_ALLOWED,
+        attestation_type=?attestation_type,
+        secure_boot=attestation_vm_config.secure_boot,
+        tpm_enabled=attestation_vm_config.tpm_enabled,
+        tpm_persisted=attestation_vm_config.tpm_persisted,
+        "Reading security profile");
+
     // Read Security Profile from VMGS
     // Currently this only includes "Key Reference" data, which is not attested data, is opaque to the
     // Underhill, and is passed to the IGVMm agent outside of the report contents.
-    let SecurityProfile { mut agent_data } = vmgs::read_security_profile(vmgs)
-        .await
-        .map_err(ErrorInner::ReadSecurityProfile)?;
+    let SecurityProfile { mut agent_data } =
+        vmgs::read_security_profile(vmgs).await.map_err(|e| {
+            Error::log_error_id(&e);
```

Comment:
I don't know much about this scenario, but I'm looking in from the outside to (a) build up my knowledge, (b) see how other engineers are approaching these problems, and (c) be an active participant in this project.

In general - I'm curious why you decided to log error IDs here. These traces have no other context (e.g. you might see "AttestationError::TypeFoo" in the logs). It seems that this would benefit from a string that adds context - e.g. so that the log consumer knows that it was an error reading the VMGS security profile.

Also, I don't think it is desirable to log type names as part of your errors. Are these as stable as you'd like? Presumably it would be better to simply have these errors format themselves into a friendly name that is stable across releases?

Reply:
The purpose of these are error IDs are to help root cause the problem and so that they can also be used for monitoring.  I would expect these enum type names not to change, but I do a agree something more stable would be better.  I think in the long term there would be a more stable error ID convention that we could use across all modules.  The goal is to identify which module the error is coming from and the specific failure scenario, this way it would speed up root causing issues and also have a way to monitor frequent failure scenarios.

Reply:
@mattkur, you're asking the right questions here. Putting the type names in the error is not the approach we want to take, and I'm working with sergio (and john) to land on a better approach here, where there is a clear separation and mapping between the Rust types and the (ideally stable) error strings / codes we log.

Reply:
I looked at the kusto traces and there is some level of hierarchy that I could potentially use in the "target" field.  I have removed the logging of the id.